subsys/bluetooth/rpc/common/serialize.h

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file
+ * @defgroup bt_rpc_serialize Bluetooth RPC serialize API
+ * @{
+ * @brief API for the Bluetooth RPC serialization.
+ */
+
+
+#ifndef SERIALIZE_H_
+#define SERIALIZE_H_
+
+#include <tinycbor/cbor.h>
+#include <net/buf.h>
+#include <sys/util.h>
+
+/** @brief Get a scratchpad item size aligned to 4-byte boundary.
+ *
+ * @param[in] size scratchpad item size
+ *
+ * @retval The scratchpad item size rounded up to the next multiple of 4.
+ */
+#define SCRATCHPAD_ALIGN(size) WB_UP(size)
+
+/** @brief Alloc the scratchpad. Scratchpad is used to store a data when decoding serialized data.
+ *
+ *  @param[in] _scratchpad Scratchpad name.
+ *  @param[in] _value Cbor value to decode. One unsigned integer will be decoded
+ *                    from this value that contains scratchpad buffer size.
+ */
+#define SER_SCRATCHPAD_ALLOC(_scratchpad, _value)                                               \
```

Comment:
That sounds reasonable.

Most of the code was created using our internal generator. `SER_SCRATCHPAD_ALLOC` is a way to allow create a scratchpad in a different way and the generator does not need to know how. That's why also `_FREE` macro call was generated, because generator does not know if the scratchpad have to be deallocated dynamically.

I think that we should delete `SER_SCRATCHPAD_FREE` macro and I have no objections regarding renaming of the `SER_SCRATCHPAD_ALLOC` macro.